PopMovies/PopMovies/Sources/Views/MovieDetails/Views/MovieDetailsView.swift

```diff
@@ -15,6 +15,13 @@ class MovieDetailsView: PMView, UIScrollViewDelegate {
     var movie: MovieItem
     var isLoadingMoreMovies = false
     let fetchSimilarMovies: () -> Void
+    var isFavorite: Bool {
+        didSet {
+            setImageButton(isFavorite)
+        }
+    }
+    var favoriteButtonSelectedAction: (_ movie: MovieItem) -> Void
+    var favoriteButtonUnselectedAction: (_ id: Int) -> Void
```

Comment:
```suggestion
    let favoriteButtonSelectedAction: (_ movie: MovieItem) -> Void
    let favoriteButtonUnselectedAction: (_ id: Int) -> Void
```

além disso, acredito que poderia fazer isso através de uma única função, apenas passando o `movie` pra sua `viewModel`:

```swift
// FavoriteMoviesViewModel
func didTapFavoriteButton(for movie: MovieItem) { 
    // Verifica se o movie está nos favoritos, caso estiver, remove, se não, adiciona
    if isFavoriteMovie { 
        removeMovie()
    } else { 
        addMovie() 
    }
}
```